.eslintrc.js

```diff
@@ -252,6 +252,26 @@ module.exports = {
     'no-native-reassign': 2,
     'no-redeclare': 2,
     'no-restricted-globals': [2, 'error', 'event', 'Animation'],
+    'no-restricted-syntax': [
+      2,
+      // Ban all TS features that don't have a direct ECMAScript equivalent.
+      {
+        'selector': 'TSEnumDeclaration',
+        'message': 'Enums are banned.',
+      },
+      {
+        'selector': 'TSModuleDeclaration',
+        'message': 'Namespaces are banned.',
+      },
+      {
+        'selector': 'TSParameterProperty',
+        'message': 'Parameter properties are banned.',
+      },
+      {
+        'selector': 'Decorator',
+        'message': 'Decorators are banned.',
+      },
+    ],
```

Comment:
Another thing I'd like to see banned is using class properties to declare types without an explicit `declare` keyword:

```js
//bad
class Foo {
  // This actually sets the value to `undefined`.
  prop: string;
}

// good
class Foo {
  declare prop: string;
}
class Foo {
  // initializing to a default is fine
  prop: string = 'default';
}

```

Reply:
Sure. Done. Please double check the selector though...it worked in my local tests but I don't have a great grasp of eslint selectors